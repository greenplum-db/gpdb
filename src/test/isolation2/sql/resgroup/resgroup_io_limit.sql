-- positive: io limit with correct syntax
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000');
SELECT io_limit FROM gp_toolkit.gp_resgroup_config WHERE groupname='rg_test_group';
DROP RESOURCE GROUP rg_test_group;

-- with space
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit=' pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000');
DROP RESOURCE GROUP rg_test_group;

-- with space
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000, wbps=1000,riops=1000,wiops=1000');
DROP RESOURCE GROUP rg_test_group;

-- with *
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='*:rbps=1000,wbps=1000,riops=1000,wiops=1000');
DROP RESOURCE GROUP rg_test_group;

-- negative: io limit with incorrect syntax
-- * must be unique tablespace
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000;*:wbps=1000');

-- tail ;
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000;');

-- tail ,
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,');

