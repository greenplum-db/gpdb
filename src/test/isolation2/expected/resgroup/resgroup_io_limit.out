-- positive: io limit with correct syntax
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000');
CREATE
SELECT io_limit FROM gp_toolkit.gp_resgroup_config WHERE groupname='rg_test_group';
 io_limit                                             
------------------------------------------------------
 pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000 
(1 row)
DROP RESOURCE GROUP rg_test_group;
DROP

-- with space
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit=' pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000');
CREATE
DROP RESOURCE GROUP rg_test_group;
DROP

-- with space
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000, wbps=1000,riops=1000,wiops=1000');
CREATE
DROP RESOURCE GROUP rg_test_group;
DROP

-- with *
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='*:rbps=1000,wbps=1000,riops=1000,wiops=1000');
CREATE
DROP RESOURCE GROUP rg_test_group;
DROP

-- negative: io limit with incorrect syntax
-- * must be unique tablespace
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000;*:wbps=1000');
ERROR:  io limit: tablespace '*' cannot be used with other tablespaces  (seg1 127.0.0.2:6001 pid=203823)

-- tail ;
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,wiops=1000;');
ERROR:  syntax error  (seg1 127.0.0.2:6001 pid=203823)

-- tail ,
CREATE RESOURCE GROUP rg_test_group WITH (concurrency=10, cpu_max_percent=10, io_limit='pg_default:rbps=1000,wbps=1000,riops=1000,');
ERROR:  syntax error  (seg1 127.0.0.2:6001 pid=203823)

